test/parallel/test-http2-util-headers-list.js

```diff
@@ -82,16 +82,12 @@ const {
     'BAR': [1]
   };
 
-  assert.deepStrictEqual(mapToHeaders(headers), [
-    [ ':path', 'abc' ],
-    [ ':status', '200' ],
-    [ 'abc', '1' ],
-    [ 'xyz', '1' ],
-    [ 'xyz', '2' ],
-    [ 'xyz', '3' ],
-    [ 'xyz', '4' ],
-    [ 'bar', '1' ]
-  ]);
+  assert.deepStrictEqual(
+    mapToHeaders(headers),
+    [ ':path_abc_:status_200_abc_1_xyz_1_xyz_2_xyz_3_xyz_4_bar_1_'
+        .replace(/_/g, '\0'),
+      8 ]
```

Comment:
using `mapToHeaders(headers).split('\0')` and keeping the second argument as an array (albeit not an array of arrays) may be easier to read